### PR TITLE
Fix tus.isSupported on iOS in Private Mode

### LINKED ISSUE #49: iOS private mode: Missing check for tus.isSupported.
I recently encountered an issue where the value of `tus.isSupported` was set to `true` but the browser didn't support tus:

The private mode of Safari on iOS doesn't allow access to the local storage ([see this thread](http://stackoverflow.com/questions/21159301/quotaexceedederror-dom-exception-22-an-attempt-was-made-to-add-something-to-st)).

Therefore it might be good to implement a check which checks whether the local storage is accessible, [something like this maybe](http://stackoverflow.com/questions/16427636/check-if-localstorage-is-available).

### PR DESCRIPTION
**Pull request by mmathys**
The transpile job wasn't executed yet, see #52

### CODE REVIEW

File: lib/browser/storage.js [hunk 1]
```diff
@@ -5,10 +5,18 @@ try {
   hasStorage = "localStorage" in window;
   // Attempt to access localStorage
   localStorage.length;
```
Comment (Acconut) [inline lib/browser/storage.js]: I believe this access could be removed now, as it's superseded by our additions.

File: lib/browser/storage.js [hunk 2]
```diff
@@ -5,10 +5,18 @@ try {
...
+
+  // Attempt to store and read entries from the local storage to detect Private
+  // Mode on Safari on iOS (see #49)
+  var test = 'tusSupport';
+  localStorage.setItem(test, test);
+  localStorage.removeItem(test);
```
Comment (mmathys) [inline lib/browser/storage.js]: When test doesn't exist in the first place you'd attempt to set and item with the value `undefined`, then an error would be thrown even when there was local storage support

Comment (mmathys) [inline lib/browser/storage.js]: Also, I have the feeling that the probability that an attribute with `tusSupport` is used by another application is very small.

Alternatively, we could use something like `'tusSupport_dHVzU3VwcG9ydA=='`

**Comment by mmathys** [comment]
Okay @Acconut. If you say so, I'll change it in a bit

**Comment by mmathys** [comment]
Didn't transpile anything yet. Didn't know how to merge your @Acconut commit back into my fork.

**Comment by Acconut** [comment]
Great, thank you! Would you also mind addressing the linting issues (see https://travis-ci.org/tus/tus-js-client/jobs/178254879#L406)?

**Comment by Acconut** [comment]
> Didn't know how to merge your @Acconut commit back into my fork.

You should be able to pull from master into your local fork using `git pull https://github.com/tus/tus-js-client.git master`. This will include the necessary commits which make the transpile step work on macOS.

**Comment by Acconut** [comment]
Has been merged in https://github.com/tus/tus-js-client/commit/19594295025114603bb7757491252316c8e167f3. Thank you for helping with this!

**Pull request closed without merging**